src/foremast/pipeline/create_pipeline_kubernetes.py

```diff
@@ -0,0 +1,196 @@
+        deploy_type = pipeline['type']
+        # Pass env in as region when getting an existing pipeline ID
+        # This is because in AWS our pipelines are "name [region]" but in k8s they are "name [env/account]"
+        pipeline_id = self.compare_with_existing(region=environment)
+
+        data = {
+            'app': {
+                'appname': self.app_name,
+                'group_name': self.group_name,
+                'repo_name': self.repo_name,
+                'deploy_type': deploy_type,
+                'region': region,
+                'environment': environment,
+                'triggerjob': self.trigger_job,
+                'run_as_user': DEFAULT_RUN_AS_USER,
+                'email': email,
+                'slack': slack,
+                'pipeline': pipeline,
+                'manifest_account_name': manifest_account
+            },
+            'id': pipeline_id
+        }
+
+        # If they specified a canary name, add it to the jinja template data
+        if 'canaryConfigName' in self.settings['pipeline']['kubernetes'].keys():
+            data['canary'] = self.create_canary_config()
+        return data
+
+    def create_canary_config(self):
+        """Creats a canary config that can be passed to the a J2 template for kayenta canaries
+        1. Checks if the pipeline.kubernetes.canaryConfigName setting was set by the user
+        2. If set, other canary configuration options like canaryAnalysisIntervalMins is added to the config
+        3. The canary config Id is resolved from Spinnaker using the name set by the user
+        It is not verified if the current app has the correct permissions to use the canary config requested.
+        """
+        kube_config = self.settings['pipeline']['kubernetes']
+        interval_minutes = kube_config['canaryAnalysisIntervalMins']
+        canary_name = kube_config['canaryConfigName']
+
+        canary = {
+            'canaryAnalysisIntervalMins': interval_minutes,
+            'canaryConfigId': self.get_canary_id(canary_name)
+        }
+
+        return canary
+
+    def get_canary_id(self, name):
+        """Finds a canary config ID matching the name passed.
+        Assumes the canary name is unique and the first match wins.
+        """
+        url = "{}/v2/canaryConfig".format(API_URL)
+        canary_response = requests.get(url, headers=self.header, verify=GATE_CA_BUNDLE, cert=GATE_CLIENT_CERT)
+
+        if not canary_response.ok:
+            raise SpinnakerPipelineCreationFailed('Pipeline for {0}: {1}'.format(self.app_name,
+                                                                                 canary_response.json()))
+
+        canary_configs = canary_response.json()
+        names = []
+        for config in canary_configs:
+            names.append(config['name'])
+            if config['name'] == name:
+                return config['id']
+
+        raise SpinnakerPipelineCreationFailed(
```

Comment:
This could probably be added to the `for` loop before as an `else` condition, like so:
```
for i in range(100):
    names.append(i)
    if i == name:
        ...
else:
    raise SpinnakerPipelineCreationFailed(...)
```